src/puppetlabs/puppetdb/query_eng/engine.clj

```diff
@@ -667,13 +667,31 @@
         [:null name]
         [field name]))))
 
+(defn merge-function-options
+  "Optionally merge call and grouping into an existing query map.
+   Alias function calls with escape-quoted function name for hsqldb compatability."
+  [m call grouping]
+
+  (if-not call
+    m
+    (let [fn-call (apply hcore/call call)
+          fn-alias (format "\"%s\"" (name (first call)))]
+      (-> m
+          (update-in [:select] conj [fn-call fn-alias])
+          (#(if grouping (assoc % :group-by (map keyword grouping)) %))))))
```

Comment:
If you used `cond->` this might be a little cleaner
```
(cond-> m
true (update-in [:select] conj [fn-call fn-alias])
grouping (assoc % :group-by (map keyword grouping))
```